server/src/main/java/org/elasticsearch/common/lucene/store/ByteArrayIndexInput.java

```diff
@@ -67,22 +67,22 @@ public long length() {
 
     @Override
     public byte readByte(long pos) throws IOException {
-        return bytes[position(pos)];
+        return bytes[offset + position(pos)];
```

Comment:
Would it be possible to have the bugfix parts here extracted to a separate PR?

Reply:
I'm on it sorry for mixing these two in the first place, couple min! :)

Reply:
Thanks!